### ISSUE TITLE
DialogFullscreen: Field warnings have insufficient colour contrast with grey background

### ISSUE THREAD

**Issue by Chris3y** [Sage/carbon#7400]
### Description

Warning messages on form inputs (when in a dialog) have insufficient contrast against their background, when in a dialog (which is usually a bit more grey than white in Sage Theme).

Important note: Axe tool is very inconsistent here and often misses this problem. You can still verify the contrast is insufficient, using browser dev tools and/or manually checking the contrast against standards (WCAG 2.1AA).

I've provided a sandbox where the problem can be verified but using Chrome or Firefox dev tools and/or [webaim-contrast-checker](https://webaim.org/resources/contrastchecker/?fcolor=BF5200FF&bcolor=F2F5F6).

### Reproduction

https://stackblitz.com/edit/parsium-carbon-starter-ebujaks2?file=src%2FApp.tsx

### Steps to reproduce

In most other areas there is a default background colour of #f8f9f9, which is a lighter grey with no contrast issues, so I’d suggest using that.

The full screen dialog background colour is set to #f2f5f6 (#f2f5f6ff / --colorsUtilityMajor025).

https://carbon.sage.com/?path=/docs/dialog-full-screen--default 

The hint text caution uses #bf5200  (#bf5200ff / --colorsSemanticCaution600), and this is a fail on the #f2f5f6 background: https://webaim.org/resources/contrastchecker/?fcolor=BF5200FF&bcolor=F2F5F6

But it will pass if used on a #f8f9f9 background.

### Carbon version

153.0.0

### React version

v18

### Relevant browsers

Chrome

### Relevant OSs

Windows

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.

**Comment by harpalsingh** [comment]
@Parsium we can't do that because there is a use case of both backgrounds still, so we need to look at the issue here on why the content on the dialog isnt in a tile, and we should ensure the color passes on the light grey as well as that use case can happen in the app on the light grey too.

**Comment by harpalsingh** [comment]
@Parsium @Chris3y will work with @ljemmo and we will bump the warning text token color slight, its .2 off, which visually wont even be noticeable but mathematically fix the contrast checker.

**Comment by ljemmo** [comment]
@harpalsingh FYI i brought the topic of the gray content area for dialogs and sidebar to the ECT as i know you were wanting to support light gray for notifications. The consensus at the time was to only support white bgs going forward as the BU reps felt that we could simplify to one color.

If you strongly disagree, i'd suggest raising it on the next ECT meeting next thursday or in the ECT slack channel. We recently added 'don't's to represent that decision: https://designsystem.sage.com/35ee2cc26/p/8314ff-dialog/t/a5b27dc0bb

**Comment by harpalsingh** [comment]
@ljemmo Yes I can bring this up in the next ECT as I was not involved in the previous decisions.

**Comment by ljemmo** [comment]
@harpalsingh and @Chris3y - the hex code for warning text was #C93E08

**Comment by harpalsingh** [comment]
For reference this will be the updated visual to compare to current version.

<img width="740" height="335" alt="Image" src="https://github.com/user-attachments/assets/def0bc43-ec65-46f2-9073-cb5bc4997928" />

**Comment by Parsium** [comment]
FE-7444
